File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.java
class Solution {
    public int[] searchRange(int[] nums, int target) {
        int[] arr= new int[2];
        int start=0;
        int end =nums.length-1;
        while(start<=end){
            int mid = start+(end-start)/2;
            if(target<nums[mid]){
                end =mid-1;
            }
            else if(target>nums[mid]){
                start= mid+1;
            }
            else{
                int x=mid, y=mid;
               while(x>=0 && nums[x]==nums[mid]){
                x--;
               }
               while(y<nums.length && nums[y]==nums[mid]){
                y++;
               }
            return new int[]{x+1,y-1};
        }
    }
        return new int[]{-1,-1};
    }
}

File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.cpp
class Solution {
public:
    int firstOccurance(vector<int>&nums,int target, int low,int high){
        int idx=-1;
        while(low<=high){
        int mid=low+(high-low)/2;
        if(nums[mid]==target){
           idx=mid;
           high=mid-1;
        }
        else if(nums[mid]>target){
            high=mid-1;
        }
        else low=mid+1;
     }
     return idx;
    }
    int lastOccurance(vector<int>&nums,int target, int low,int high){
        int idx=-1;
        while(low<=high){
            int mid=low+(high-low)/2;
            if(nums[mid]==target){
                idx=mid;
                low=mid+1;
            }
            else if(nums[mid]<target){
                low=mid+1;
            }
            else high=mid-1;
        }
        return idx;
    }

    vector<int> searchRange(vector<int>& nums, int target) {
        vector<int>v(2);
        int n=nums.size();
        v[0]=firstOccurance(nums,target,0,n-1);
        v[1]=lastOccurance(nums,target,0,n-1);
        return v;
    }
};